sklearn/tests/test_metadata_routing.py

```diff
@@ -73,14 +73,16 @@ def assert_request_equal(request, dictionary):
 def record_metadata(obj, method, **kwargs):
     """Utility function to store passed metadata to a method."""
     if not hasattr(obj, "_records"):
-        setattr(obj, "_records", dict())
+        obj._records = {}
```

Comment:
I don't understand your impulse to change `setattr` :D I also don't use `{}` since to me it's a vague syntax. `{}` is an empty dict, but `{1}` is a `set`.

Reply:
I think the change I made is more idiomatic than the previous code, which makes it easier to read. `setattr` for me implies that the attribute name is dynamic, which is when you really need it. You wouldn't do `def __init__(self, arg): setattr(self, "arg", arg)`, would you :D 

About the dict: I see your point but I think that `{}` is so common in Python that there is no danger of mistaking it for a set. Furthermore:

```
%timeit _ = {}
21.1 ns ± 0.054 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)

%timeit _ = dict()
78.9 ns ± 2.19 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)
```
;)

All that said, if you want me to revert the change, I'm happy to do that.